services/types.go

```diff
@@ -244,34 +273,52 @@ type PlacementService interface {
 	Delete() error
 }
 
-// PlacementOptions is the interface for placement options
+// PlacementOptions is the interface for placement options.
 type PlacementOptions interface {
 	// LooseRackCheck enables the placement to loose the rack check
-	// during instance replacement to achieve full ownership transfer
+	// during instance replacement to achieve full ownership transfer.
 	LooseRackCheck() bool
+
+	// SetLooseRackCheck sets LooseRackCheck.
 	SetLooseRackCheck(looseRackCheck bool) PlacementOptions
 
 	// AllowPartialReplace allows shards from the leaving instance to be
 	// placed on instances other than the new instances in a replace operation
 	AllowPartialReplace() bool
+
+	// SetAllowPartialReplace sets AllowPartialReplace.
 	SetAllowPartialReplace(allowPartialReplace bool) PlacementOptions
 
-	// IsSharded describes whether a placement needs to be sharded
-	// when set to false, no specific shards will be assigned to any instance
+	// IsSharded describes whether a placement needs to be sharded,
+	// when set to false, no specific shards will be assigned to any instance.
 	IsSharded() bool
+
+	// SetIsSharded sets IsSharded.
 	SetIsSharded(sharded bool) PlacementOptions
 
-	// Dryrun will try to perform the placement operation but will not persist the final result
+	// Dryrun will try to perform the placement operation but will not persist the final result.
 	Dryrun() bool
+
+	// SetDryrun sets whether the Dryrun value.
 	SetDryrun(d bool) PlacementOptions
 
-	// IsMirrored sets whether the shard distribution should be mirrored
+	// IsMirrored returns whether the shard distribution should be mirrored
 	// to support master/slave model.
 	IsMirrored() bool
+
+	// SetIsMirrored sets IsMirrored.
 	SetIsMirrored(m bool) PlacementOptions
 
-	// InstrumentOptions is the options for instrument
+	// ShouldKeepSnapshots returns whether the placement should keep all the snapshots.
+	ShouldKeepSnapshots() bool
```

Comment:
nit: `IsStagedPlacement` might be a better name?

Reply:
Sure